Sming/Components/rboot/component.mk

```diff
@@ -21,37 +21,61 @@ endif
 # => APP
 
 # rBoot options, overwrite them in the projects Makefile-user.mk
-CONFIG_VARS				+= RBOOT_BIG_FLASH RBOOT_TWO_ROMS RBOOT_RTC_ENABLED RBOOT_GPIO_ENABLED RBOOT_GPIO_SKIP_ENABLED
-RBOOT_BIG_FLASH			?= 1
-RBOOT_TWO_ROMS			?= 0
+CONFIG_VARS				+= RBOOT_RTC_ENABLED RBOOT_GPIO_ENABLED RBOOT_GPIO_SKIP_ENABLED
 RBOOT_RTC_ENABLED		?= 0
 RBOOT_GPIO_ENABLED		?= 0
 # RBOOT_GPIO_SKIP_ENABLED and RBOOT_GPIO_ENABLED cannot be used at the same time.
 RBOOT_GPIO_SKIP_ENABLED	?= 0
 
 ifeq ($(RBOOT_GPIO_ENABLED)$(RBOOT_GPIO_SKIP_ENABLED),11)
-$(error "Cannot enable RBOOT_GPIO_ENABLED and RBOOT_GPIO_SKIP_ENABLED at the same time)
+$(error Cannot enable RBOOT_GPIO_ENABLED and RBOOT_GPIO_SKIP_ENABLED at the same time)
 endif
 
 ### ROM Addresses ###
-# The parameter below specifies the location of the second rom.
-# This parameter is used only when RBOOT_BIG_FLASH = 1 
-# BOOT_ROM1_ADDR = 0x200000
+# Make sure that your ROM slots and SPIFFS slot(s) do not overlap!
+
+# Loation of first ROM (default is sector 2 after rboot and rboot config sector)
+RBOOT_ROM0_ADDR			?= 0x002000
+
+# The parameter below specifies the location of the second ROM.
+# You need a second slot for any kind of firmware update mechanism.
+# Leave empty if you don't need a second ROM slot.
+RBOOT_ROM1_ADDR			?=
 
 # The parameter below specifies the location of the GPIO ROM.
-# This parameter is used only when RBOOT_GPIO_ENABLED = 1
-# If you use two SPIFFS make sure that this address is minimum
-# RBOOT_SPIFFS_1 + SPIFF_SIZE 
-# BOOT_ROM2_ADDR = 0x310000
+# It is only used when RBOOT_GPIO_ENABLED = 1
+# Note that setting this parameter will only configure rboot.
+# The Sming build system does not create a ROM for this slot.
+RBOOT_ROM2_ADDR			?=
+
+ifeq ($(RBOOT_GPIO_ENABLED),0)
+ifneq ($(RBOOT_ROM2_ADDR),)
+$(warning RBOOT_GPIO_ENABLED is 0, RBOOT_ROM2_ADDR will be ignored)
+RBOOT_ROM2_ADDR :=
+endif
+endif
+
+# determine number of roms to generate
+ifneq ($(RBOOT_ROM1_ADDR),)
```

Comment:
As RBOOT_TWO_ROMS has been removed as a CONFIG_VAR, it would be helpful to add DEBUG_VARS += RBOOT_TWO_ROMS so we can check the logic.